covalent/_results_manager/results_manager.py

```diff
@@ -49,59 +57,105 @@ def get_result(
+    )
+    result._status = lattice_record.status
+    result._error = error
+    result._inputs = inputs
+    result._start_time = lattice_record.started_at
+    result._end_time = lattice_record.completed_at
+    result._lattice.transport_graph = transport_graph
+    result._lattice.args = inputs["args"]
+    result._lattice.kwargs = inputs["kwargs"]
+    result._result = output
+    app_log.warning(f"Transport graph sorted: {transport_graph.get_topologically_sorted_graph()}")
+    return result
+
+
+def _get_result_from_db(
+    db: DataStore, dispatch_id: str, wait: bool = False, status_only: bool = False
 ) -> Result:
 
     """
     Internal function to get the results of a dispatch from a file without checking if it is ready to read.
 
     Args:
+        db: The DataStore object which yields the SQLAlchemy session that stores results.
         dispatch_id: The dispatch id of the result.
-        results_dir: The directory where the results are stored in dispatch id named folders.
         wait: Whether to wait for the result to be completed/failed, default is False.
 
     Returns:
-        result_object: The result from the file.
+        result_object: The result object constructed from the DB lattice record.
 
     Raises:
-        RuntimeError: If the result is found but is not ready to read yet.
-        FileNotFoundError: If the result is not found.
+        MissingLatticeRecordError: If the result is not found.
     """
 
-    results_dir_expanded = str(Path(results_dir).expanduser().resolve())
-    result_dir = os.path.join(results_dir_expanded, f"{dispatch_id}")
-
     while True:
+        with Session(db.engine) as session:
+            lattice_record = (
+                session.query(Lattice).where(Lattice.dispatch_id == dispatch_id).first()
+            )
         try:
-            with open(os.path.join(result_dir, "result.pkl"), "rb") as f:
-                result = pickle.loads(f.read())
-
-            if not wait:
-                return result
-            elif result.status in [
-                Result.COMPLETED,
-                Result.POSTPROCESSING_FAILED,
-                Result.PENDING_POSTPROCESSING,
-                Result.FAILED,
-                Result.CANCELLED,
+            if not lattice_record:
+                raise MissingLatticeRecordError
+            elif not wait:
+                return lattice_record.status if status_only else result_from(lattice_record)
+            elif lattice_record.status in [
+                str(Result.COMPLETED),
+                str(Result.FAILED),
+                str(Result.CANCELLED),
+                str(Result.POSTPROCESSING_FAILED),
+                str(Result.PENDING_POSTPROCESSING),
             ]:
-                return result
+                return lattice_record.status if status_only else result_from(lattice_record)
+
         except (FileNotFoundError, EOFError, _pickle.UnpicklingError):
             if wait:
                 continue
```

Comment:
I think we should start here to figure out why everything is failing again

Reply:
The ["Hello World" workflow](https://covalent.readthedocs.io/en/latest/getting_started/index.html#hello-covalent) now fails.

One problem is that `Lattice.function` is now [serialized when the workflow is first constructed](https://github.com/AgnostiqHQ/covalent/blob/88a56179b45f7400496a09510bd2530d8a23906b/covalent/_workflow/lattice.py#L100) and never deserialized by the server (we should probably add tests to ensure that the Covalent server never calls `TransportableObject.get_deserialized()`). Thus [this line](https://github.com/AgnostiqHQ/covalent/blob/88a56179b45f7400496a09510bd2530d8a23906b/covalent/_results_manager/results_manager.py#L102) no longer works because the retrieved `function` is a `TransportableObject`. The ordinary `Lattice` constructor tries to extract the function name by calling `self.function.__name__`. 

Suggestion: Don't directly invoke the Lattice constructor (which is expecting an ordinary function). After retrieving all the attributes from a `lattice_record`, create an attribute dictionary, create an dummy Lattice (e.g. with the identity function), and set its attributes via `lattice.__dict__ = [attribute dictionary]`

Reply:
This is now fixed! Thank you @cjao for your suggestion. The next issue is related to postprocessing the sublattice and lattice results in my sample workflow. I imagine it has something to do with the code expecting a TransportableObject but getting a scalar value.

Reply:
[This](https://github.com/AgnostiqHQ/covalent/blob/378131b777de2185c70c52a483cdfa54c3952875/covalent/_results_manager/result.py#L728) is the problematic line. `Result.result` tries to deserialize the result and should never be called by the server. Instead use `Result.encoded_result`.

Reply:
Yessir! These are changed. Still getting some related errors tho
<img width="743" alt="Screen Shot 2022-07-17 at 2 16 50 PM" src="https://user-images.githubusercontent.com/17885542/179419355-da4ecb16-47b7-4dcf-b976-af373b7d233f.png">
:

Reply:
Can you confirm that the node outputs in the transport graph are TransportableObjects before starting post-processing?